docs/tutorials/create-an-ethereum-processing-squid.md

```diff
@@ -0,0 +1,455 @@
+* **`@derivedFrom`** - signals the field will not be persisted on the database, it will rather be derived
+* **type references** (i.e. `from: Owner`) - establishes a relation between two entities
+* **`@index`** - signals thatt his field should be indexed by the database. Very useful for incresing performance on fields queried often.
+
+The template already has automatically generated TypeScript classes for this schema definition. They can be found under `src/model/generated`.
+
+Whenever changes are made to the schema, new TypeScript entity classes have to be generated, and to do that you'll have to run the `codegen` tool:
+
+```bash
+make codegen
+```
+
+## ABI Definition and Wrapper
+
+Subsquid offers support for automatically building TypeScript type-safe interfaces and decoding functions for EVM smart contracts, thanks to the `evm-typegen` tool.
+
+First, it is necessary to obtain the definition of the Application Binary Interface (ABI). This can be obtained in the form of a JSON file, which will be imported into the project. In the context of this example, you can find the Exosama collection smart contract (a modified ERC-721) at [this address](https://github.com/RaekwonIII/eth-workshop/raw/master/src/abi/exo.json).
+
+1. It is advisable to copy the JSON file in the `src/abi` subfolder.
+2. To automatically generate TypeScript interfaces from an ABI definition, and decode event data, simply run this command from the project's root folder
+
+```bash
+npx squid-evm-typegen --abi src/abi/exo.json --output src/abi/exo.ts
+```
+
+The `abi` parameter points at the JSON file previously created, and the `output` parameter is the name of the file that will be generated by the command itself.
+
+This command will automatically generate a TypeScript file named `exo.ts`, under the `src/abi` subfolder, that defines data interfaces to represent output of the EVM events defined in the ABI, as well as a mapping of the functions necessary to decode these events (see the `events` dictionary in the aforementioned file).
+
+## Define and Bind Event Handler(s)
+
+The Subsquid SDK provides users with the `EvmBatchProcessor`, that connects to the [Subsquid archive](/overview) to get chain data and apply custom transformation. It will index from the starting block, until the end block (if these are set in the configuration), or until new data is added to the chain.
+
+The processor exposes methods to "subscribe" to EVM logs or smart contract function calls. These methods can be configured by specifying the EVM log contract address, and the signature of the EVM event, for example. The actual data processing is then started by calling the `.run()` function. This will start generating requests to the Archive for *batches* of the data specified in the configuration, and will trigger a *handler function* every time a batch is returned by the Archive itself.
```

Comment:
"*handler function*" handlers are not supported by evm-processor

Reply:
What would you call the second argument of `.run()`?

Reply:
Would "the callback function passed as second parameter" work for you?

I'm changing the title of the paragraph, "Define and Bind Event Handler(s)" which I think is even worse.

Reply:
I use "the batch handler"

Reply:
it's more like "process function"